README.md

```diff
@@ -119,6 +132,57 @@ authenticate scheme.
 fetch: (url: RequestInfo, options?: Object) => Promise<Response>
 ```
 
+## Developing
+
+### Prerequisites
+
+This library assumes you have [node](https://nodejs.org/en/) >= v7.10.1  and
+[yarn](https://yarnpkg.com/) 0.24.6 installed.  It may work with earlier
+versions, but that hasn't been tested thus far.
+
+### Setting up the development environment
+
+```sh
+$ git clone https://github.com/solid/solid-auth-client.git
+$ cd solid-auth-client
+$ yarn
+$ yarn build # build the library and UMD bundle
+$ yarn test # run the code formatter, linter, and test suite
+$ yarn test:dev # just run the tests in watch mode
+```
+
+### Building the demo app
+
+```sh
+$ cp .env.demo.example .env.demo && $EDITOR .env.demo # configure the demo app
+$ yarn start:demo
+```
+
+#### Configuration
+
+The demo app is configurable via the `.env.demo` file.  The important fields are:
+
+- IDP_SELECT_URI: URI for the popup-based IDP select app.  When testing
+  locally, this will be something like 'http://localhost:XXXX/idp-select.html'
+- CALLBACK_URI: URI for the popup-based callback app.  When testing locally,
+  this will be something like 'http://localhost:XXXX/idp-callback.html'
+
+### Building the popup app
```

Comment:
Perhaps explain why I would want to do this?

Reply:
Maybe "app" is a bit confusing, in the sense that "demo app" and "popup app" mean different things. The popup is a mandatory component, the demo is just an external app.

Reply:
And the title is about building, but we are actually running it.